File: python/menstrual app/menstrual_app_driver.py
from menstrual_app import cycle_dates_calculation

while True:
	print("MENSTRUAL TRACKER APP")
	user_date = input("Enter the first day of your last period(YYYY-MM-DD): ")
	while True:
		try:
			cycle_length = int(input("Enter your cycle length in days: "))
			break
		except ValueError:
			print("Days must be a whole number ")

	try:
		info = cycle_dates_calculation(user_date, cycle_length)
		print("Menstrual Cycle information: ")
		print(f"start_date: {info['start_date']}")
		print(f"next_period_date: {info['next_period']}")
		print(f"ovulation_day: {info['ovulation_day']}")
		print(f"Fertile Window: {info['fertile_window'][0]} to {info['fertile_window'][1]}")
		break
	except ValueError:
		print("Invalid date format. Please try again using the YYYY-MM-DD format ")

File: javascript/menstrual app/MenstrualAppDriver.js
const prompt = require("prompt-sync")();
const { calculateCycleDates } = require("./menstrualApp");

console.log("MENSTRUAL TRACKER APP");

let startDateStr;
let cycleLength;

while (true) {
    startDateStr = prompt("Enter the first day of your last period (YYYY-MM-DD): ");
    const date = new Date(startDateStr);

    if (!isNaN(date)) {
        break;
    } else {
        console.log("Invalid date format. Please use YYYY-MM-DD.");
    }
}

while (true) {
    const input = prompt("Enter your cycle length in days: ");
    cycleLength = parseInt(input);

    if (!isNaN(cycleLength)) {
        break;
    } else {
        console.log("Number of Days must be a whole number.");
    }
}

try {
    const info = calculateCycleDates(startDateStr, cycleLength);

    console.log("Menstrual Cycle Information:");
    console.log("Start Date:", info.startDate.toDateString());
    console.log("Next Period:", info.nextPeriod.toDateString());
    console.log("Ovulation Day:", info.ovulationDay.toDateString());
    console.log("Fertile Window:", info.fertileWindow[0].toDateString(), "to", info.fertileWindow[1].toDateString());
} catch (error) {
    console.log("Error:", error.message);
}
